modin/core/dataframe/algebra/groupby.py

```diff
@@ -290,6 +291,21 @@ def caller(
             )
         assert axis == 0, "Can only groupby reduce with axis=0"
 
+        # The bug only occurs in the case of Categorical 'by', so we might want to check whether any of
+        # the 'by' dtypes is Categorical before going into this branch, however triggering 'dtypes'
+        # computation if they're not computed may take time, so we don't do it
```

Comment:
[`modin_frame._dtypes`](https://github.com/modin-project/modin/blob/73bf061900f4e797f58ed02037de12a2c670d2a1/modin/core/dataframe/pandas/dataframe/dataframe.py#L153-L164) attribute is `None` most of the time, [its computation](https://github.com/modin-project/modin/blob/73bf061900f4e797f58ed02037de12a2c670d2a1/modin/core/dataframe/pandas/dataframe/dataframe.py#L166-L188) requires calling MapReduce and actual data materialization. I don't think we want to trigger this on every `groupby(sort=False)`